test/gtest/common/mem_buffer.cc

```diff
@@ -95,12 +95,17 @@ void mem_buffer::set_device_context()
     static __thread bool device_not_set = true;
 
     if (device_not_set) {
+        try {
 #if HAVE_CUDA
-        CUDA_CALL(cudaSetDevice(0), "set device context");
+            CUDA_CALL(cudaSetDevice(0), "set device context");
 #endif
 #if HAVE_ROCM
-        ROCM_CALL(hipSetDevice(0));
+            ROCM_CALL(hipSetDevice(0));
 #endif
+        } catch (...) {
+            UCS_TEST_ABORT("Unable to set device context");
```

Comment:
UCS_TEST_ABORT throws
maybe just do cudaSetDevice w/o CUDA_CALL/ROCM_CALL wrappers
or in main.cc
try {
  set_device_context
} catch {
  UCS_TEST_MESSAGE << failed to set gpu device context
}

Reply:
Fixed this.